casbah/src/main/scala/akka/contrib/persistence/mongodb/CasbahPersistenceJournaller.scala

```diff
@@ -45,49 +45,33 @@ class CasbahPersistenceJournaller(driver: CasbahMongoDriver) extends MongoPersis
       .map(driver.deserializeJournal)
   }
 
-  import collection.immutable.{ Seq => ISeq }
-  private[this] def doBatchJournalAppend(writes: ISeq[AtomicWrite], journal: MongoCollection)(implicit ec: ExecutionContext): ISeq[Try[Unit]] = {
+  import collection.immutable.{ Seq => ISeq }  
+  private[this] def doBatchAppend(writes: ISeq[AtomicWrite], collection: MongoCollection)(implicit ec: ExecutionContext): ISeq[Try[Unit]] = {
     val batch = writes.map(write => Try(driver.serializeJournal(Atom[DBObject](write, driver.useLegacySerialization))))
     
     if (batch.forall(_.isSuccess)) {
-      val bulk = journal.initializeOrderedBulkOperation
+      val bulk = collection.initializeOrderedBulkOperation
       batch.collect { case scala.util.Success(ser) => ser } foreach bulk.insert
       bulk.execute(writeConcern)
       batch.map(t => t.map(_ => ()))
     } else { // degraded performance, can't batch
-      batch.map(_.map { serialized =>
-        journal.insert(serialized)(identity, writeConcern)
-        //if (driver.realtimeEnablePersistence) realtime.insert(serialized)(identity, writeConcern)
-      }.map(_ => ()))
-    }
-  }
-
-  private[this] def doBatchRealtimeAppend(writes: ISeq[AtomicWrite])(implicit ec: ExecutionContext): ISeq[Try[Unit]] = {
-    val batch = writes.map(write => Try(driver.serializeJournal(Atom[DBObject](write, driver.useLegacySerialization))))
-    if (batch.forall(_.isSuccess)) {
-      val bulk = realtime.initializeOrderedBulkOperation
-      batch.collect { case scala.util.Success(ser) => ser } foreach bulk.insert
-      bulk.execute(writeConcern)
-
-      batch.map(t => t.map(_ => ()))
-    } else { // degraded performance, can't batch
-      batch.map(_.map { serialized => realtime.insert(serialized)(identity, writeConcern) }.map(_ => ()))
+      batch.map(_.map { serialized => collection.insert(serialized)(identity, writeConcern) }.map(_ => ()))
     }
   }
 
   private[mongodb] override def batchAppend(writes: ISeq[AtomicWrite])(implicit ec: ExecutionContext): Future[ISeq[Try[Unit]]] = {
     val batchFuture = Future {
       if (driver.useSuffixedCollectionNames) {
-        writes.groupBy(_.persistenceId).flatMap {
-          case (pid, writeSeq) => doBatchJournalAppend(writeSeq, driver.journal(pid))
+        writes.groupBy(w => driver.getSuffixFromPersistenceId(w.persistenceId)).flatMap {
+          case (pid, writeSeq) => doBatchAppend(writeSeq, driver.journal(pid))
```

Comment:
Ouch... Actually, this does not work, the code is now wrong as we now group by **suffix**, so we call driver.journal(**suffix**) instead of calling driver.journal(**persistenceId**)

Of course, I could create some `driver.journalFromSuffix(suffix)` method, but this would introduce another level of complexity: if one `persistenceId` correspond for sure to only one *suffix*, at the opposite one *suffix* may correspond to several `persistenceId`s.

Plus, the code in the `groupBy` method calls `driver.getSuffixFromPersistenceId` method, so, for <u>**each**</u> `AtomicWrite` we instantiate the class (provided by the user) extending `CanSuffixCollectionNames` trait and call its `getSuffixFromPersistenceId` method.

I wonder which one is more efficient:
- grouping by persistenceId and may be bulk inserting several times in the same collection if its *suffix* correspond to several `persistenceId`s
- calling  `driver.getSuffixFromPersistenceId` method for <u>**each**</u> `AtomicWrite`

I guess the former is more efficient and less complex to implement. And as it is little code to update, I revert my previous changes right now (the end of the day is coming here ;-)) and we'll discuss this later...

Notice that even if code is wrong, tests seem good because `getSuffixFromPersistenceId(persistenceId)` simply returns `persistenceId`. Here *suffix* **equals** `persistenceId` :-/